src/furrent.cpp

```diff
@@ -0,0 +1,55 @@
+#include "furrent.hpp"
+
+#include <fstream>
+#include <iostream>
+#include <sstream>
+
+#include "bencode_parser.hpp"
+#include "torrent_manager.hpp"
+
+using namespace fur;
+
+// Create constructor
+Furrent::Furrent() :
+         _downloads(std::vector<manager::TorrentManager>()),
+        _index(0) {
+}
+
+// Add torrent to downloads
+void Furrent::add_torrent(const std::string& path) {
+  // Read all file data located in the path
+  std::ifstream file(path);
+  std::string content;
+  if (file) {
+    std::ostringstream ss;
+    ss << file.rdbuf();
+    content = ss.str();
+  } else {
+    // Trow exception invalid path
+    throw std::invalid_argument("fur::Furrent::add_torrent: invalid path");
```

Comment:
Not necessarily (I suppose), can also be a `missing permissions` error and such. Although I'm being overly critical here.

Reply:
We have to check if we can use `exception` in the project, but the comment is useful for me thanks